### ISSUE TITLE
Support for cursors

### ISSUE THREAD

**Issue by ryanheise** [simolus3/sqlite3.dart#49]
This is needed to iterate over a large result set without preloading the entire set into memory. `ResultSet.iterator` could be made to work (maybe `BidirectionalIterator` could also be useful), but it currently isn't implemented like a database cursor.

**Comment by simolus3** [comment]
I think `BidirectionalIterator` wouldn't really work without caching in memory, so if you care about not consuming more memory than necessary I'd just expose it as an `Iterator` probably.

**Comment by simolus3** [comment]
I've added an `PreparedStatement.selectCursor` that steps through rows one by one in 8f1cdb3d4d5dd2acaeb7b568b726f336e8a68fbf.

You would use it like this:

```dart
final stmt = db.prepare('SELECT foo FROM bar');
final cursor = stmt.selectCursor();

while (cursor.moveNext()) {
  // use row with cursor.current
}
```

I opted to create a new method only returning an `Iterator` because `Iterable` typically implies that you can iterate over them multiple times. That's not really the case for an prepared statement, so it doesn't feel like the right interface to expose here.

Let me know if that sounds like it would work for you, I'll release it to pub then.

**Comment by ryanheise** [comment]
Thanks for the super fast response! I will try it out today.

In terms of the API, I will leave that to you. Although it seems to me that if  `selectCursor()` is the iterator, then `PreparedStatement` is the iterable (but only if you had a specialised `PreparedSelectStatement`). i.e. because the semantics of `Iterable` is that each time you iterate over it, it invokes `iterator` afresh.

**Comment by ryanheise** [comment]
Just tested it and it's working perfectly :+1:

**Comment by simolus3** [comment]
> Although it seems to me that if  `selectCursor()` is the iterator, then `PreparedStatement` is the iterable (but only if you had a specialised `PreparedSelectStatement`)

That's a good point. I had considered something like this, but in the end we'd have to prevent multiple iterators interleaving, so it will always be kind of a weird `Iterable`.

I've released the iterator in `1.2.0`, we can always come up with a more fancy API later :)
